Makefile

```diff
@@ -0,0 +1,542 @@
+# Copyright 2018  Google & GKE.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+PROTOCOL ?= gs:
+REGISTRY_PATH ?= kubernetes-release/release/v1.13.2
+REGISTRY ?= ${PROTOCOL}//${REGISTRY_PATH}
+
+build:
+	mkdir -p build
+	mkdir -p build/server
+	mkdir -p build/src
+	mkdir -p build/manifests
+
+_output:
+	mkdir -p _output
+	mkdir -p _output/server/kubernetes
+
+vendor:
+	mkdir -p vendor
+
+build/%.tar.gz.sha512: build
+	# echo -n does not work below because the problem is the newline in the sha512 file not the echo.
+	gsutil cp ${REGISTRY}/$*.tar.gz.sha512 build; \
+	mv $@ $@.tmp; \
+	echo " build/$*.tar.gz" >> $@.tmp; \
+	tr -d '\n' < $@.tmp > $@; \
+	rm $@.tmp
+
+build/%.tar.gz: build/%.tar.gz.sha512
+	@echo Fetching from ${REGISTRY}/$*.tar.gz
+	gsutil cp ${REGISTRY}/$*.tar.gz build; \
+	sha512sum -c $<
+	echo "Success"
+
+build/server: build build/kubernetes-server-linux-amd64.tar.gz
+	# build/server is the location of the unpacked build/kubernetes-server-linux-amd64.tar.gz
+	tar xvfz build/kubernetes-server-linux-amd64.tar.gz -C build/server
+
+build/src: build build/server
+	# build/server is the location of the unpacked build/build/server/kubernetes/kubernetes-src.tar.gz
+	# Cannot have a rule for kubernetes-src.tar.gz as there is no shasum for it
+	tar xvfz build/server/kubernetes/kubernetes-src.tar.gz -C build/src
+
+_output/ccm-vendor.tar.gz: build/src
```

Comment:
Is there a way to use `go list` to find the subset of `vendor/` to include?

Reply:
Why do you need only vendor? Should we be gather all source code linked into ccm?

Reply:
Good question, waiting to see what happens with #75. I would like to do this more cleanly but only if we come back to this.